test/commands/interactive.test.js

```diff
@@ -0,0 +1,117 @@
+const assert = require('assert');
+const sinon = require('sinon');
+const path = require('path');
+const Promise = require('bluebird');
+
+const logger = require('../../lib/utils/logger');
+const interactive = require('../../lib/commands/interactive');
+const repl = require('../../lib/testLauncher/repl');
+const texts = require('../../lib/texts');
+const {authContext} = require('../../lib/context');
+
+const stubbed = {};
+
+describe('interactive command', () => {
+	before(() => {
+		process.env.REPL_IPC_PORT = 'fake';
+	});
+
+	beforeEach(() => {
+		sinon.stub(logger);
+		stubbed.authContext = sinon.stub(authContext, 'isInteractiveSession').returns(true);
+	});
+	afterEach(() => {
+		sinon.restore();
+		stubbed.authContext.restore();
+	});
+	it('should not start another repl instance if one is already running', async() => {
+		const replSessionEnded = sinon.stub(texts, 'replSessionEnded');
+
+		let startRepl;
+		let replCalls = 0;
+
+		return new Promise(async resolve => {
+			startRepl = sinon.stub(repl, 'startRepl').callsFake(async() => {
+				await interactive();
+				++replCalls;
+				assert.equal(replCalls, 1, 'Second call is ignored');
+				resolve();
+			});
+			await interactive();
+		}).then(async() => {
```

Comment:
Why just not await promise above without additional chains or mocha will not close process?

Reply:
Because I need to simulate two parallel processes calling `interactive()` in overlap, i.e. test calling `interactive()` while we are still awaiting the first `interactive()` to finish. Look closer at the location of `resolve()`.